File: 动画更新masonary/动画更新masonary/ViewController.m
//
//  ViewController.m
//  动画更新masonary
//
//  Created by 廖佩志 on 2018/9/7.
//  Copyright © 2018年 廖佩志. All rights reserved.
//

#import "ViewController.h"
#import <Masonry.h>
@interface ViewController ()
- (IBAction)btnClick:(id)sender;
@property(nonatomic,strong)UIView *animationView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (void)updateViewConstraints{
    [self.animationView mas_updateConstraints:^(MASConstraintMaker *make) {
        make.center.mas_equalTo(self.view);
        //初始化为1
        make.width.height.mas_equalTo(1).priorityLow();
        //
        make.width.mas_equalTo(200);
        make.height.mas_equalTo(200);
    }];
    [super updateViewConstraints];
}

- (IBAction)btnClick:(id)sender {
    UIView *aniView = [[UIView alloc]init];
    [self.view addSubview:aniView];
    self.animationView = aniView;
    [aniView mas_updateConstraints:^(MASConstraintMaker *make) {
        make.center.mas_equalTo(self.view);
        //初始化为1
        make.width.height.mas_equalTo(1).priorityLow();
        //
        make.width.mas_equalTo(200);
        make.height.mas_equalTo(200);
    }];
    
//    [aniView mas_updateConstraints:^(MASConstraintMaker *make) {
//        make.center.mas_equalTo(self.view);
//        //初始化为1
//        make.width.height.mas_equalTo(1).priorityLow();
//        //
//        make.width.mas_equalTo(200);
//        make.height.mas_equalTo(200);
//    }];

    [self.view setNeedsUpdateConstraints];
    
    // update constraints now so we can animate the change
    [self.view updateConstraintsIfNeeded];
    
    [UIView animateWithDuration:0.4 animations:^{
        [self.view layoutIfNeeded];
    }];
}
@end

File: testForDidSet/testForDidSet/ViewController.swift
//
//  ViewController.swift
//  testForDidSet
//
//  Created by 廖佩志 on 2018/11/30.
//  Copyright © 2018 廖佩志. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var a: String = "" {
        didSet{
            a = "123"
            a = "456"
        }
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        a = "1"
        print(a)
        let view1 = UIView(frame: CGRect(x: 100, y: 100, width: 100, height: 100))
        view1.backgroundColor = .red
        view.addSubview(view1)
        let view2 = UIView(frame: CGRect(x: 200, y: 100, width: 100, height: 100))
        view2.backgroundColor = .green
        print(view2.bounds)
        view2.bounds = view2.bounds.insetBy(dx: -50, dy: -50)
        print(view2.bounds)
        view.addSubview(view2)


        // Do any additional setup after loading the view, typically from a nib.
    }



}

